src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/DraftReferralDTO.kt

```diff
@@ -44,6 +46,8 @@ data class DraftReferralDTO(
         maximumRarDays = referral.maximumRarDays,
         desiredOutcomesIds = referral.desiredOutcomesIDs,
         serviceUser = ServiceUserDTO.from(referral.serviceUserCRN, referral.serviceUserData),
+        serviceProvider = ServiceProviderDTO.from(contract.serviceProvider),
```

Comment:
_suggestion, not a blocker_
The full chain is `referral.intervention.dynamicFrameworkContract.serviceProvider`; would it make sense to add a service to look up contract or provider by a referral?